### PR TITLE
cylc globalcfg: upgrade to ISO syn, doc, back comp.

### LINKED ISSUE #802: siterc: configurable event hooks and timeouts
We should be able to set site defaults for ~~batch sizes~~, timeouts and event hooks.

These would be brought into the suite from the site/user config in much the same way as polling intervals.

### LINKED ISSUE #1266: cylc global config: upgrade to ISO syntax, document, and fix back. comp.
Someone (OK, me) missed updating the site and user configuration parsing in
parsec - at the moment options like 'submission polling intervals' are:
 * actually used as seconds, not minutes as they used to be (hence old values will give different results now!)
 * not compatible with new ISO 8601 syntax

The relevant file is `lib/cylc/cfgspec/globalcfg.py`. It needs work to bring it into line with
`lib/cylc/cfgspec/suite.py` - e.g. porting the 'float_list' type specification for the `polling intervals`
options to use `interval_minutes_list`.

Once these are fixed, the documentation also needs to be updated.

### PR DESCRIPTION
**Pull request by kaday**
Close #802. Close #1266.

Tested locally and works for me.

No test added to the test battery yet - unsure how to do this therefore investigating.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@kaday Search for `CYLC_CONF_PATH` in other tests. The environment variable provides a way to override the search locations for global configuration files.

**Comment by kaday** [comment]
Test added.
@benfitzpatrick please review

File: doc/siterc.tex [hunk 1]
```diff
@@ -103,8 +103,8 @@ \subsubsection{execution polling intervals}
 Multipliers can be used as shorthand as in the example below.
 
 \begin{myitemize}
-\item {\em type:} list of float minutes with optional multipliers
```
Comment (benfitzpatrick) [inline doc/siterc.tex]: See e.g. https://github.com/cylc/cylc/blob/master/doc/suiterc.tex#L1400-L1404

**Comment by kaday** [comment]
@benfitzpatrick agreed and done, all of the appropriate example syntax examples have been updated.

File: lib/cylc/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -30,6 +41,50 @@
 
 "Cylc site and user configuration file spec."
 
+interval_parser = DurationParser()
+
+def _coerce_interval( value, keys, args, back_comp_unit_factor=1 ):
```
Comment (benfitzpatrick) [inline lib/cylc/cfgspec/globalcfg.py]: Aren't these the same as in `suite.py`? Could they just be renamed without the underscore and imported directly from there?

File: tests/jobscript/05-global-config/conf/global.rc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+execution polling intervals = PT0.2M, PT0.1M
+submission polling intervals = PT0.2M, PT0.1M
+[task messaging]
+   connection timeout = PT20S
+   retry interval = PT9S
```
Comment (benfitzpatrick) [inline tests/jobscript/05-global-config/conf/global.rc]: I think there isn't a test for this bit?

Comment (benfitzpatrick) [inline tests/jobscript/05-global-config/conf/global.rc]: I'm wrong!

**Comment by kaday** [comment]
@benfitzpatrick removed def's and _co... , added deprecate for both "in seconds".

File: doc/siterc.tex (hunk 1 above)
Comment (benfitzpatrick) [inline doc/siterc.tex]: I mean the type = ISO 8601 something something bit

**Comment by benfitzpatrick** [comment]
One remaining comment about the documented type - otherwise good to go.

**Comment by kaday** [comment]
@benfitzpatrick hopefully done and rebased

**Pull request merged**